3.0/Dockerfile

```diff
@@ -1,66 +1,90 @@
+ADD tiger.tar.gz data/
+
+# Runtime Dependencies
+RUN apt-get install -y apache2 php php-pgsql libapache2-mod-php php-pear php-db \
+php-intl python-pip libboost-python-dev gdal-bin python-gdal \
+postgresql-9.5-postgis-2.2 postgresql-contrib-9.5 \
+libbz2-dev libz-dev unzip sudo
+
+# TODO Possibly Unnecessary Dependencies?
+RUN apt-get install -y osmosis
 RUN pip install osmium
 
-# Apache configure
-COPY local.php /app/src/build/settings/local.php
+# Cleanup runtime after dependecy install
+RUN apt-get clean
+RUN rm -rf /var/lib/apt/lists/*
+RUN rm -rf /tmp/* /var/tmp/*
+
+# Apache
+COPY local.php build/settings/local.php
 COPY nominatim.conf /etc/apache2/sites-enabled/000-default.conf
 
-# Load initial data
-ARG PBF_DATA=http://download.geofabrik.de/north-america-latest.osm.pbf
-ARG TIGER_DATA=https://nominatim.org/data/tiger2017-nominatim-preprocessed.tar.gz
-RUN curl -L -f $PBF_DATA --create-dirs -o /app/src/data.osm.pbf
-RUN curl -L -f $TIGER_DATA --create-dirs -o /app/src/data/tiger2017.tar.gz
-RUN curl http://www.nominatim.org/data/country_grid.sql.gz > /app/src/data/country_osm_grid.sql.gz
-RUN tar -xzf /app/src/data/tiger2017.tar.gz -C /app/src/data/
+# Postgres
+RUN echo "host all  all    0.0.0.0/0  trust" >> /etc/postgresql/9.5/main/pg_hba.conf
+RUN echo "listen_addresses='*'" >> /etc/postgresql/9.5/main/postgresql.conf
+RUN sed -i -e"s/^max_connections = 100.*$/max_connections = 1000/" /etc/postgresql/9.5/main/postgresql.conf
+
+# nominatim user
+# TODO Do we need the nominatim user?  This is docker...
+RUN useradd nominatim && \
+chown -R nominatim:nominatim /app/src
+
+# initial state of Postgres DB
 RUN service postgresql start && \
-    sudo -u postgres psql postgres -tAc "SELECT 1 FROM pg_roles WHERE rolname='nominatim'" | grep -q 1 || sudo -u postgres createuser -s nominatim && \
-    sudo -u postgres psql postgres -tAc "SELECT 1 FROM pg_roles WHERE rolname='www-data'" | grep -q 1 || sudo -u postgres createuser -SDR www-data && \
-    sudo -u postgres psql postgres -c "DROP DATABASE IF EXISTS nominatim" && \
-    useradd -m -p password1234 nominatim && \
-    chown -R nominatim:nominatim ./src && \
-    sudo -u nominatim ./src/build/utils/setup.php --osm-file /app/src/data.osm.pbf --all --import-tiger-data --threads 64 && \
-    sudo -u nominatim ./src/build/utils/setup.php --create-functions --enable-diff-updates --create-partition-functions && \
-    service postgresql stop 
+    sudo -u postgres createuser -s nominatim && \
+    sudo -u postgres createuser -SDR www-data && \
+    sudo -u nominatim /app/src/build/utils/setup.php --osm-file /app/src/data/data.osm.pbf --all --import-tiger-data --threads 64 && \
+    sudo -u nominatim /app/src/build/utils/setup.php --create-functions --enable-diff-updates --create-partition-functions && \
+    service postgresql stop
 
 EXPOSE 5432
-EXPOSE 8080
 
-COPY start.sh /app/start.sh
-CMD /app/start.sh
+# TODO Is this necessary to expose in production?
+# EXPOSE 8080
```

Comment:
I think ports do need to be exposed for a docker image to listen on them. Even if we switched this to plain old 80, wouldn't you have to list it? 

Also, the URLs that are baked into GIS service have this port number (i.e. GIS is calling http://nominatim.holmansv.io:8080/reversegeocode/blablabla). If we were running multiple containers on a single host, it would be advantageous to run these on different ports, right?